packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+    init(): void {
+        // Send and show proxy-view
+        this.actionDispatcher.dispatch(SendProxyViewAction.create(this));
+        this.actionDispatcher.dispatch(ShowProxyViewAction.create());
+        this.patcher = this.patcherProvider.patcher;
+        this.renderings = new Map;
+        this.positions = new Map;
+    }
+
+    protected initializeContents(containerElement: HTMLElement): void {
+        // Use temp for initializing oldContentRoot
+        const temp = document.createElement("div");
+        this.currHTMLRoot = this.patcher(temp, <div />);
+        containerElement.appendChild(temp);
+    }
+
+    protected onBeforeShow(containerElement: HTMLElement, root: Readonly<SModelRoot>, ...contextElementIds: string[]): void {
+        // TODO: could be useful?
+    }
+
+    //////// Main methods ////////
+
+    // !!! TODO: might be a useful addition to save absolute coords in SKNode, not my task but also required here
+
+    /** Updates the proxy-view. */
+    update(model: SGraph, ctx: SKGraphModelRenderer): void {
+        if (!this.proxyViewEnabled) {
+            this.currHTMLRoot = this.patcher(this.currHTMLRoot, <div />);
+            return;
+        } else if (!this.currHTMLRoot) {
+            return;
+        } else if (!ctx.depthMap) {
+            // Create a new depthmap if otherwise unused
+            ctx.depthMap = DepthMap.init(model);
+        }
+
+        const canvasWidth = model.canvasBounds.width;
+        const canvasHeight = model.canvasBounds.height;
+        const scroll = model.scroll;
+        const zoom = model.zoom;
+        const root = model.children[0] as SKNode;
+        this.currHTMLRoot = this.patcher(this.currHTMLRoot,
+            <svg style={
+                {
+                    width: canvasWidth.toString(), height: canvasHeight.toString(), // Set size to whole canvas
+                    pointerEvents: "none" // Make click-through
+                }
+            }>
+                {...this.createAllProxies(root, ctx, canvasWidth, canvasHeight, scroll, zoom)}
+            </svg>);
+    }
+
+    /** Returns the proxy rendering for all of currRoot's off-screen children and applies logic, e.g. clustering. */
+    private createAllProxies(root: SKNode, ctx: SKGraphModelRenderer, canvasWidth: number, canvasHeight: number, scroll: Point, zoom: number): VNode[] {
+        // Iterate through nodes starting by root, check if node is: 
+        // (partially) in bounds -> no proxy, check children
+        // out of bounds         -> proxy
+
+        const depthMap = ctx.depthMap;
+        const viewport = ctx.viewport;
+        if (!depthMap || !viewport) {
+            // Not yet initialized
+            return [];
+        }
+
+        //// Initial nodes ////
+        const { offScreenNodes, onScreenNodes } = this.getOffAndOnScreenNodes(root, ctx);
+
+        //// Apply filters ////
+        const filteredOffScreenNodes = this.applyFilters(offScreenNodes, onScreenNodes);
```

Comment:
this reads a little confusing, the on screen nodes are passed in as a parameter for the filters and the result is a subset of the original off screen nodes, correct?